analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/BranchingProcessor.cs

```diff
@@ -41,7 +41,10 @@ protected override ProgramState[] Process(SymbolicContext context, T operation)
         var state = PreProcess(context.State, operation);
         if (BoolConstraintFromOperation(state, operation, context.IsLoopCondition, context.VisitCount) is { } constraint)
         {
-            return state.SetOperationConstraint(context.Operation, constraint).ToArray();    // We already know the answer from existing constraints
+            state = state.SetOperationConstraint(context.Operation, constraint);    // We already know the answer from existing constraints
+            return context.VisitCount > 1
```

Comment:
I would delete the comment above here - I'm not sure what it means with my current knowledge - and I would add a comment to line 45.
// We are branching for both conditions only for the second loop
of something similar.

Reply:
We do not branch here. We are just applying fuzzy logic to NumberConstraints.